include/SSVOpenHexagon/Core/HexagonGame.hpp

```diff
@@ -160,13 +160,46 @@ class HexagonGame
     template <typename T, typename... TArgs>
     T runLuaFunction(const std::string& mName, const TArgs&... mArgs)
     {
-        return Utils::runLuaFunction<T, TArgs...>(lua, mName, mArgs...);
+        try {
+            return Utils::runLuaFunction<T, TArgs...>(lua, mName, mArgs...);
+        } 
+        catch(std::runtime_error& mError)
+        {
+            std::cout << "[Lua] Runtime error on \""
+                      << mName
+                      << "\" with level \""
+                      << levelData -> name
+                      << "\": \n"
+                      << ssvu::toStr(mError.what()) << "\n"
+                      << std::endl;
+            if (!Config::getDebug())
+            {
+                goToMenu(false, true);
+            }
+        }
+        return T();
     }
 
     template <typename T, typename... TArgs>
     void runLuaFunctionIfExists(const std::string& mName, const TArgs&... mArgs)
     {
-        Utils::runLuaFunctionIfExists<T, TArgs...>(lua, mName, mArgs...);
+        try {
+            Utils::runLuaFunctionIfExists<T, TArgs...>(lua, mName, mArgs...);
+        }
+        catch(std::runtime_error& mError)
+        {
+            std::cout << "[Lua] Runtime error on \""
```

Comment:
```suggestion
            std::cout << "[runLuaFunctionIfExists] Runtime error on \""
```

Reply:
Could you amend this a bit so that if you are going to report it for "IfExists" specifically, you could get rid of "(optional function)". That way it's not so redundant.